pkg/vault/contracts/test/PoolHooksMock.sol

```diff
@@ -63,6 +63,10 @@ contract PoolHooksMock is BaseHooks {
     address private _specialSender;
     uint256[] private _newBalancesRaw;
 
+    // Bool created because in some tests the test file is used as router and does not implement getSender.
+    bool public shouldStoreSavedSender;
```

Comment:
Consider reversing the logic here, given that all the tests need to set it to true (which seems to be the most common case). If you called it "shouldIgnoreSavedSender", it would default to false, and you wouldn't have to set it true everywhere. (Unless that means you'd have to set it false in even more places, of course.)